### PR TITLE
Create new projects for template files and schema files, renaming the exiting files to their own modules.

### PR DESCRIPTION
**Pull request by ms-teli**
Changes include:
1. Move template to a sharable assembly that is using .NET Standard 2.0.
2. Move the data model of the Measurement and the MeasurementGroup to a sharable assembly.
3. Move and update the unit tests.

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/IotJsonPathContentTemplateTests.cs [hunk 1]
```diff
@@ -38,17 +39,15 @@ public class IotJsonPathContentTemplateTests
         [FileData(@"TestInput/data_IotHubPayloadExample.json")]
         public void GivenTemplateAndSingleValidToken_WhenGetMeasurements_ThenSingleMeasurementReturned_Test(string eventJson)
         {
-            var evt = EventDataTestHelper.BuildEventFromJson(eventJson);
-            var token = new EventDataWithJsonBodyToJTokenConverter().Convert(evt);
-
+            var token = JsonConvert.DeserializeObject<JToken>(eventJson);
```
Comment (dustinburson) [inline test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/IotJsonPathContentTemplateTests.cs:42]: Why the change from the EventDataTestHelper Event to a JToken?

Comment (ms-teli) [reply test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/IotJsonPathContentTemplateTests.cs:42]: Thanks for reviewing.

Because otherwise, the EventHub dependencies would still be required in this assembly which is not quite necessary for the measurement models. These tests were primarily for the template so this should not reduce the testing coverage by skipping the converters. And in the original package, we still have UTs to cover the conversions.

Comment (dustinburson) [reply test/Microsoft.Health.Fhir.Ingest.Template.UnitTests/IotJsonPathContentTemplateTests.cs:42]: Makes sense. Thanks for clarifying.
